main.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"fmt"
+	gp "hangman/gameplay"
+	ui "hangman/user_input"
+	"strings"
+)
+
+// main function
+func main() {
+	var guesses int
+	var targetWord string
+	var guessedChars []string
+	var remainingWord []string
+
+	guesses = gp.GetDifficulty()
+	numLeft := guesses
+	targetWord = gp.GetTargetWord()
+	remainingWord = strings.Split(targetWord, "")
+
+	fmt.Printf("You will have: %v guesses \n", guesses)
+	fmt.Println(targetWord)
+
+	for i := 0; i < guesses; i++ {
+		guessedLetter := ui.GuessLetter(targetWord, guessedChars)
+		guessedChars = append(guessedChars, guessedLetter)
+
+		if strings.Contains(strings.Join(remainingWord, ""), guessedLetter) {
```

Comment:
There doesn't seem to be a clear separation between the code here in main and the code in your individual function. Your main function should control the flow while the functions control the behavior